File: js/top_ten_scorers.mjs
import readline from "readline"
const reader = readline.createInterface(process.stdin)

const players = []

reader.on("line", line => {
  let [team, name, games, points] = line.split(",")
  if (games >= 15) {
    players.push({ name, team, ppg: points / games })
  }
})
reader.on("close", () => {
  const topTen = players.sort((p1, p2) => p2.ppg - p1.ppg).slice(0, 10)
  for (let { name, team, ppg } of topTen) {
    console.log(
      `${name.padEnd(22)}${team.padEnd(4)}${ppg.toFixed(2).padStart(8)}`
    )
  }
})

File: python/top_ten_scorers.py
import sys
from heapq import nlargest
from operator import itemgetter

players = []

for line in sys.stdin:
    team, name, games, points = line.strip().split(',')
    if int(games) >= 15:
        players.append((name, team, int(points)/int(games)))

for name, team, ppg in nlargest(10, players, key=itemgetter(2)):
    print(f'{name:22}{team:4}{ppg:8.2f}')
